alerting_alert_rule.go

```diff
@@ -0,0 +1,95 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// AlertRule represents a Grafana Alerting Alert Rule.
+type AlertRule struct {
+	Annotations  map[string]string `json:"annotations,omitempty"`
+	Conditions   string            `json:"conditions"`
+	Data         []*AlertQuery     `json:"data"`
+	ExecErrState ExecErrState      `json:"execErrState"`
+	FolderUID    string            `json:"folderUid"`
+	ID           int64             `json:"id,omitempty"`
+	Labels       map[string]string `json:"labels,omitempty"`
+	NoDataState  NoDataState       `json:"noDataState"`
+	OrgID        int64             `json:"orgId"`
+	RuleGroup    string            `json:"ruleGroup"`
+	Title        string            `json:"title"`
+	UID          string            `json:"uid,omitempty"`
+	Updated      string            `json:"updated,omitempty"`
+	ForDuration  time.Duration     `json:"for"`
+	Provenance   string            `json:"provenance"`
+}
+
+type AlertQuery struct {
+	DatasourceUID     string
+	Model             interface{}
+	QueryType         string
+	RefID             string
+	RelativeTimeRange RelativeTimeRange
+}
+
+type ExecErrState string
+type NoDataState string
+
+const (
+	ErrOK       ExecErrState = "OK"
+	ErrError    ExecErrState = "Error"
+	ErrAlerting ExecErrState = "Alerting"
+	NoDataOk    NoDataState  = "OK"
+	NoData      NoDataState  = "NoData"
+	NoDataErr   NoDataState  = "Error"
+)
+
+// RelativeTimeRange represents the time range for an alert query.
+type RelativeTimeRange struct {
+	From time.Duration `json:"from"`
+	To   time.Duration `json:"to"`
+}
+
+// AlertRule fetches a single alert rule, identified by its UID.
+func (c *Client) AlertRule(uid string) (AlertRule, error) {
+	path := fmt.Sprintf("/api/v1/provisioning/alert-rules/%s", uid)
+	result := AlertRule{}
+	err := c.request("GET", path, nil, nil, &result)
+	if err != nil {
+		return AlertRule{}, fmt.Errorf("alert rule with uid %s not found", uid)
```

Comment:
```suggestion
		return AlertRule{}, err
```
If it's a 404, the `err` will already say that the resource wasn't found in the message. For cases where the error isn't a 404, (unauthorized, can't reach server, etc) we don't want to return a potentially misleading message.